### PR TITLE
Documentation on test name flags.

### PR DESCRIPTION
**Pull request by jgraham**

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5699/)

File: docs/_writing-tests/general-guidelines.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ your `test-topic` is probably too broad!)
 In csswg-test, the file names should be unique within the repository,
 regardless of where they are in the directory structure.
 
+### HTTPS
+
+By default tests are served over plain http. If a test requires https
+it must be given a filename ending `.https.ext`
```
Comment (annevk) [inline docs/_writing-tests/general-guidelines.md]: Does .https need to be before .ext or can you also have https.sub.ext?

File: docs/_writing-tests/general-guidelines.md [hunk 2]
```diff
@@ -28,6 +28,12 @@ your `test-topic` is probably too broad!)
...
+e.g. `test-secure.https.html`. For more details see the documentation
```
Comment (annevk) [inline docs/_writing-tests/general-guidelines.md]: commas around e.g.

File: docs/_writing-tests/file-names.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+These flags must be the last element in the filename before the
+extension e.g. `foo-manual.html` will indicate a manual test, but
+`foo-manual-other.html` will not.
+
+<dl>
+  <dt>`-manual`
+  <dd><p>Indicates that a test is a non-automated test.
+  <dt>`-support`
+  <dd><p>Indicates that a file is not a test but a support file.
```
Comment (annevk) [inline docs/_writing-tests/file-names.md]: Should we note that this is not needed if the resource is in a resources/ directory?

File: docs/_writing-tests/file-names.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+</dl>
+
+### Test Features
+
+<dl>
+  <dt>`.https`
+  <dd><p>Indicates that a test is loaded over https
+  <dt>`.sub`
+  <dd><p>Indicates that a test uses the server-side substitution
+         feature.
```
Comment (annevk) [inline docs/_writing-tests/file-names.md]: Can we cross-reference this?

File: docs/_writing-tests/file-names.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  <dd><p>Indicates that a file is not a test but a support file.
+  <dt>`-visual`
+  <dd><p>Indicates that a file is a visual test.
+</dl>
+
+### Test Features
+
+<dl>
+  <dt>`.https`
+  <dd><p>Indicates that a test is loaded over https
```
Comment (annevk) [inline docs/_writing-tests/file-names.md]: s/https/HTTPS/

File: docs/_writing-tests/file-names.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+         feature.
+  <dt>`.window`
+  <dd><p>(js files only) Indicates that the file generates a test in
+  which it is run in a Window environment.
+  <dt>`.worker`
+  <dd><p>(js files only) Indicates that the file generates a test in
+  which it is run in a decicated worker environment.
+  <dt>`.any`
+  <dd><p>(js files only) Indicates that the file generates tests in
+  which it is run in Window and decicated worker environments.
```
Comment (annevk) [inline docs/_writing-tests/file-names.md]: These would also be nice to cross-reference.

**Review by annevk — COMMENTED** [review]
Couple questions and nits. Looks good overall.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/225946838)

# Lint #

Passed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/225946841)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 1d199f809a549d7b3910bd2213f16e162640df8d
Using browser at version BuildID 20170425101537; SourceStamp a30dc237c3a600a5231f2974fc2b85dfb5513414
Starting 10 test iterations
No tests run.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/225946847)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 1d199f809a549d7b3910bd2213f16e162640df8d

File: docs/_writing-tests/general-guidelines.md (hunk 1 above)
Comment (gsnedders) [inline docs/_writing-tests/general-guidelines.md]: It should be unordered, AFAIK.

Comment (jgraham) [inline docs/_writing-tests/general-guidelines.md]: Yes, it's unordered, but explaining that well is hard.

File: docs/_writing-tests/file-names.md (hunk 6 above)
Comment (jgraham) [inline docs/_writing-tests/file-names.md]: Not sure what you want to cross-reference to here.

File: docs/_writing-tests/file-names.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  <dd><p>Indicates that a test is loaded over https
+  <dt>`.sub`
+  <dd><p>Indicates that a test uses the server-side substitution
+         feature.
+  <dt>`.window`
+  <dd><p>(js files only) Indicates that the file generates a test in
+  which it is run in a Window environment.
+  <dt>`.worker`
+  <dd><p>(js files only) Indicates that the file generates a test in
+  which it is run in a decicated worker environment.
```
Comment (sideshowbarker) [inline docs/_writing-tests/file-names.md]: s/decicated/dedicated/

File: docs/_writing-tests/file-names.md (hunk 6 above)
Comment (sideshowbarker) [inline docs/_writing-tests/file-names.md]: s/decicated/dedicated/

Comment (sideshowbarker) [inline docs/_writing-tests/file-names.md]: >> These would also be nice to cross-reference.
> Not sure what you want to cross-reference to here.

*dedicated worker environment* I assume

**Review by sideshowbarker — APPROVED** [review]
LGTM modulo the typos and things @annevk already raised

File: docs/_writing-tests/file-names.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+The test filename is significant in determining the type of test it
+contains, and enabling specific optional features. This page documents
+the various flags avaiable and their meaning.
+
+### Test Type
+
+These flags must be the last element in the filename before the
+extension e.g. `foo-manual.html` will indicate a manual test, but
+`foo-manual-other.html` will not.
```
Comment (gsnedders) [inline docs/_writing-tests/file-names.md]: I'd explicitly note they're mutually exclusive, and test features aren't.

**Review by gsnedders — APPROVED** [review]

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260172731)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260172732)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260172739)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260172740)

# Sauce (MicrosoftEdge) #

**Pull request merged**